### ISSUE TITLE
How to generate a word_w2v.txt file？

### ISSUE THREAD

**Issue by bassyess** [salman-h-khan/PL-ZSD_Release#11]
For my own data set, know the tag name, how do we generate the file?

**Comment by shafin5** [comment]
You can get word vectors from here:
https://radimrehurek.com/gensim/models/word2vec.html

For my experience, applying L2 normalization after extracting those vectors works better. You can use the following code to apply L2 normalization.
https://www.dropbox.com/s/y7c03sr3vfdzry1/SynC.zip?dl=0&file_subpath=%2FSynC%2Fcodes%2Fget_class_signatures.m

**Comment by tao0420** [comment]
Do you train the word2vec model or use some pretrained one with the abstracted phrase label for MSCOCO?

**Comment by shafin5** [comment]
No training was done for the word2vec model. Only pre-trained one is used with the class name of MSCOCO, not even any caption or phrase label annotation.

**Comment by zhongxiangzju** [comment]
Thanks for your  suggestions. @tao0420 
I tried to generate word embeddings for each label in coco dataset using gensim, but got different results.
```
import numpy as np
import gensim.downloader 
print(list(gensim.downloader.info()['models'].keys()))
# ['fasttext-wiki-news-subwords-300', 'conceptnet-numberbatch-17-06-300', 'word2vec-ruscorpora-300', 'word2vec-google-news-300', 'glove-wiki-gigaword-50', 'glove-wiki-gigaword-100', 'glove-wiki-gigaword-200', 'glove-wiki-gigaword-300', 'glove-twitter-25', 'glove-twitter-50', 'glove-twitter-100', 'glove-twitter-200', '__testing_word2vec-matrix-synopsis']
word_vectors = gensim.downloader.load('word2vec-google-news-300')
person_embedding = word_vectors['person']
person_embedding  = person_embedding / np.linalg.norm(person_embedding)
print(person_embedding)
# 0.1208263, -0.1084009, 0.00755164, 0.07369547, -0.06384084, 0.07026777 ...
```

which is different with the first row in  ./PL-ZSD_Release/MSCOCO/word_w2v.txt.
`0.092629, 0.013665, 0.037897, 0.034125, 0.015237, 0.034970 ...`

**Comment by shafin5** [comment]
I suspect the pre-trained w2v model could be different. Also, the normalization process may have differences. That could be the reason.
